tests/sa_test.go

```diff
@@ -0,0 +1,188 @@
+// +build sa
+
+/*
+ * Copyright (C) 2019 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy ofthe License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specificlanguage governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package tests
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/spf13/viper"
+
+	"github.com/skydive-project/skydive/config"
+	"github.com/skydive-project/skydive/contrib/objectstore/subscriber"
+	g "github.com/skydive-project/skydive/gremlin"
+)
+
+const securityAdvisorConfigTemplate = `---
+objectstore:
+  endpoint: http://127.0.0.1:9000
+  region: local
+  bucket: bucket
+  access_key: user
+  secret_key: password
+`
+
+func setupSecurityAdvisor(c *TestContext) error {
+	cfg := viper.New()
+
+	analyzers := config.GetStringSlice("analyzers")
+	if len(analyzers) == 0 {
+		return errors.New("Missing analyzers config")
+	}
+
+	f, err := ioutil.TempFile("", "skydive_agent")
+	if err != nil {
+		return fmt.Errorf("failed to create configuration file: %s", err)
+	}
+	if _, err = f.Write([]byte(fmt.Sprintf(securityAdvisorConfigTemplate, analyzers[0]))); err != nil {
```

Comment:
There is nothing to substitute in the template so it generates an invalid configuration file

Reply:
fixed